docs/iris/src/whatsnew/aggregate_directory.py

```diff
@@ -0,0 +1,183 @@
+# (C) British Crown Copyright 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+import datetime
+import os
+import sys
+import re
+import argparse
+import warnings
+from operator import itemgetter
+from distutils import version
+
+# Regular expressions: CONTRIBUTION_REGEX matches the filenames of
+# contribution snippets. It is split into three sections separated by _
+# 0. String for the category. 1. ISO8601 date. 2. String for the feature name.
+# RELEASE_REGEX matches the directory names, returning the release.
+CONTRIBUTION_REGEX = re.compile(r'(?P<category>.*)\_(?P<isodate>\d{4}-\d{2}-\d{2})\_(?P<summary>.*)\.txt$')
```

Comment:
NB. I don't think you need the backslash characters before the underscore characters - i.e. `\_` -> `_`.